omelet-core/src/main/java/omelet/mail/Email.java

```diff
@@ -41,340 +34,371 @@
+                timeoutPropertyKey = "mail.pop3.timeout";
+                props.setProperty("mail.pop3.socketFactory.class", "javax.net.ssl.SSLSocketFactory");
+                props.setProperty("mail.pop3.socketFactory.fallback", "false");
+                break;
+        }
+
+        props.setProperty(hostPropertyKey, host);
+        props.setProperty(portPropertyKey, port);
+        session = Session.getInstance(props, new Authenticator() {
+            @Override
+            protected PasswordAuthentication getPasswordAuthentication() {
+                return new PasswordAuthentication(userName, password);
+            }
+        });
+        connectToStore(protocol.toString().toLowerCase(), session);
+    }
+
+    private void connectToStore(String protocol, Session session) {
+        Store store;
+
+        try {
+            store = session.getStore(protocol);
+            store.connect(userName, password);
+            folder = store.getFolder(folderName);
+        } catch (NoSuchProviderException e) {
+            LOGGER.error(e);
+        } catch (MessagingException e) {
+            LOGGER.error(e);
+        }
+    }
+
+    /***
+     * Builder class to build email objects
+     *
+     * @author kapil
+     *
+     */
+    public static class Builder {
+
+        private String host;
+        private String port;
+        private String userName;
+        private String password;
+        private MailProtocol protocol = MailProtocol.POP3;
+
+        public Builder setHost(String host) {
+            this.host = host;
+            return this;
+        }
+
+        public Builder setPort(String portNo) {
+            this.port = portNo;
+            return this;
+        }
+
+        public Builder setUserName(String userName) {
+            this.userName = userName;
+            return this;
+        }
+
+        public Builder setPassword(String password) {
+            this.password = password;
+            return this;
+        }
+
+        public Builder setProtocol(MailProtocol protocol) {
+            this.protocol = protocol;
+            return this;
+        }
+
+        public Email build() {
+            return new Email(this);
+        }
+    }
+
+    /***
+     * Return format of email Message
+     */
+    @Override
+    public String getMailFormat(Message msg) {
+        String format = null;
+
+        try {
+            format = msg.getContentType();
+        } catch (MessagingException e) {
+
+            LOGGER.error(e);
+        }
+        return format;
+    }
+
+    private enum EMAIL_FILTER {
+        TO_ADDR, FROM_ADD, SUBJECT
+    }
+
+    /***
+     * Return List of Message filter by Subject,From_ADD,To_ADDR
+     * Element 0 is the newest one!!
+     *
+     * @param emailFilter enum
+     * @param filterText
+     *            :text present in Subject of email
+     * @return list of messages
+     */
+    private List<Message> getMessages(EMAIL_FILTER emailFilter, String filterText) {
```

Comment:
Please change the method name to getEmails() to be consistent across other methods